dts/arm/raspberrypi/rpi_pico/rp2040.dtsi

```diff
@@ -429,6 +429,14 @@
 		io-channels = <&adc 4>;
 		status = "disabled";
 	};
+
+	boot_flash: flash@10000000 {
```

Comment:
If we put this in 'unit address' order, this helps to highlight that something isn't right about this approach. Similarly, a quick grep of the codebase suggests that `flash@unitaddress` is always used to refer to some non-volatile storage, not to something stored at that location.